Source/JavaScriptCore/runtime/JSGlobalObjectFunctions.cpp

```diff
@@ -256,102 +256,102 @@ static bool isInfinity(const CharType* data, const CharType* end)
 
 // See ecma-262 6th 11.8.3
 template <typename CharType>
-static double jsBinaryIntegerLiteral(const CharType*& data, const CharType* end)
+static double jsBinaryIntegerLiteral(std::span<const CharType>& data)
 {
     // Binary number.
-    data += 2;
-    const CharType* firstDigitPosition = data;
+    data = data.subspan(2);
+    auto* firstDigitPosition = data.data();
```

Comment:
Use a `span` instead? It can remember both the position and how much of the data there is there for less overflow risk.

Reply:
Ok